tests/xoto3/dynamodb/write_versioned/api2_test.py

```diff
@@ -0,0 +1,99 @@
+from random import randint
+
+import pytest
+
+from xoto3.dynamodb.exceptions import ItemNotFoundException
+from xoto3.dynamodb.write_versioned import (
+    ItemTable,
+    VersionedTransaction,
+    create_or_update,
+    update_existing,
+    update_if_exists,
+    versioned_transact_write_items,
+)
+
+from .conftest import mock_next_run
+
+
+def _randkey():
+    return dict(id="".join((str(randint(0, 9)) for i in range(30))))
+
+
+def test_single_table_helpers(integration_test_id_table):
+    itable = ItemTable(integration_test_id_table.name, item_name="Steve")
+
+    rand_key = _randkey()
+    # random because it's actually writing to the table
+
+    def transact(vt):
+        vt = itable.put(dict(rand_key, foo="bar"))(vt)
```

Comment:
I wonder if it would help steer people in a good direction if we used `flow` here.

Reply:
i considered it, but honestly i don't want to imply in our docs that you need `returns` to make good use of `xoto3`.

Reply:
(and we definitely can't do it in the tests because `returns` is not a dependency)